cluster.go

```diff
@@ -595,8 +595,12 @@ func (c *clusterState) slotRandomNode(slot int) (*clusterNode, error) {
 	if len(nodes) == 0 {
 		return c.nodes.Random()
 	}
-	n := rand.Intn(len(nodes))
-	return nodes[n], nil
+	for _, idx := range rand.Perm(len(nodes)) {
+		if !nodes[idx].Failing() {
+			return nodes[idx], nil
+		}
+	}
+	return c.nodes.Random()
```

Comment:
Nice catch - thanks :+1: 

Does it make sense to try random node here? It will result in MOVED error and will re-try the command again on the same failing nodes.

I think we should:
- remove `c.nodes.Random()` and just use first/random failing node as we did previously
- add an optimization for `len(nodes) == 1` since `rand.Perm` is somewhat expensive

```go
if len(nodes) == 1 {
    return nodes[0]
}
```